test/unit/index.test.ts

```diff
@@ -0,0 +1,161 @@
+  ['Long',                               'reason'],
+  ['Map',                                'reason'],
+  ['MaxKey',                             'reason'],
+  ['MinKey',                             'reason'],
+  ['ObjectId',                           'reason'],
+  ['Timestamp',                          'reason'],
+  ['ChangeStreamCursor',                 'reason'],
+  ['ObjectID',                           'reason'],
+  // errors
+  ['MongoBulkWriteError',                'reason'],
+  ['MongoAPIError',                      'reason'],
+  ['MongoAWSError',                      'reason'],
+  ['MongoBatchReExecutionError',         'reason'],
+  ['MongoChangeStreamError',             'reason'],
+  ['MongoCompatibilityError',            'reason'],
+  ['MongoCursorExhaustedError',          'reason'],
+  ['MongoCursorInUseError',              'reason'],
+  ['MongoDecompressionError',            'reason'],
+  ['MongoDriverError',                   'reason'],
+  ['MongoError',                         'reason'],
+  ['MongoExpiredSessionError',           'reason'],
+  ['MongoGridFSChunkError',              'reason'],
+  ['MongoGridFSStreamError',             'reason'],
+  ['MongoInvalidArgumentError',          'reason'],
+  ['MongoKerberosError',                 'reason'],
+  ['MongoMissingCredentialsError',       'reason'],
+  ['MongoMissingDependencyError',        'reason'],
+  ['MongoNetworkError',                  'reason'],
+  ['MongoNetworkTimeoutError',           'reason'],
+  ['MongoNotConnectedError',             'reason'],
+  ['MongoParseError',                    'reason'],
+  ['MongoRuntimeError',                  'reason'],
+  ['MongoServerClosedError',             'reason'],
+  ['MongoServerError',                   'reason'],
+  ['MongoServerSelectionError',          'reason'],
+  ['MongoSystemError',                   'reason'],
+  ['MongoTailableCursorError',           'reason'],
+  ['MongoTopologyClosedError',           'reason'],
+  ['MongoTransactionError',              'reason'],
+  ['MongoUnexpectedServerResponseError', 'reason'],
+  ['MongoWriteConcernError',             'reason'],
+  // classes
+  ['AbstractCursor',                     'reason'],
+  ['Admin',                              'reason'],
+  ['AggregationCursor',                  'reason'],
+  ['CancellationToken',                  'reason'],
+  ['ChangeStream',                       'reason'],
+  ['ClientSession',                      'reason'],
+  ['Collection',                         'reason'],
+  ['Db',                                 'reason'],
+  ['FindCursor',                         'reason'],
+  ['GridFSBucket',                       'reason'],
+  ['ListCollectionsCursor',              'reason'],
+  ['ListIndexesCursor',                  'reason'],
+  ['Logger',                             'reason'],
+  ['MongoClient',                        'reason'],
+  // global promise setter
+  ['Promise',                            'global promise setter'],
```

Comment:
was the intention to use the string values for some sort of more detailed testing? I see we're only using ts-node, can we clarify the intention behind the "reason"? if we aren't going to use the other string, I'd suggest breaking up the exports into separately named sets or plain arrays and then testing the union or individual members as desired; currently the map approach seems unnecessarily complicated

Reply:
Yea I can change this to an array, I set up the map preparing for some discussion on how to organize the tests, but I think its fine how we currently have it, it's a pretty simple test of key existence